staging/src/k8s.io/dynamic-resource-allocation/structured/allocator_test.go

```diff
@@ -1361,6 +1421,334 @@ func TestAllocator(t *testing.T) {
+					device(device2,
+						map[resourceapi.QualifiedName]resource.Quantity{
+							"memory": resource.MustParse("2Gi"),
+						},
+						map[resourceapi.QualifiedName]resourceapi.DeviceAttribute{
+							"driverVersion": {VersionValue: ptr.To("2.0.0")},
+						},
+					),
+					device(device3,
+						map[resourceapi.QualifiedName]resource.Quantity{
+							"memory": resource.MustParse("1Gi"),
+						},
+						map[resourceapi.QualifiedName]resourceapi.DeviceAttribute{
+							"driverVersion": {VersionValue: ptr.To("1.0.0")},
+						},
+					),
+				),
+			),
+			node: node(node1, region1),
+
+			expectResults: []any{allocationResult(
+				localNodeSelector(node1),
+				deviceAllocationResult(req0, driverA, pool1, device1, false),
+				deviceAllocationResult(req1SubReq1, driverA, pool2, device3, false),
+			)},
+		},
+		"prioritized-list-subrequests-with-constraints-ref-sub-request": {
+			prioritizedList: true,
+			claimsToAllocate: objects(
+				claimWithRequests(claim0,
+					[]resourceapi.DeviceConstraint{
+						{
+							Requests:       []string{req0, req1SubReq0},
+							MatchAttribute: &versionAttribute,
+						},
+					},
+					request(req0, classA, 1, resourceapi.DeviceSelector{
+						CEL: &resourceapi.CELDeviceSelector{
+							Expression: fmt.Sprintf(`device.capacity["%s"].memory.compareTo(quantity("8Gi")) >= 0`, driverA),
+						}},
+					),
+					requestWithPrioritizedList(req1,
+						subRequest(subReq0, classA, 1, resourceapi.DeviceSelector{
+							CEL: &resourceapi.CELDeviceSelector{
+								Expression: fmt.Sprintf(`device.capacity["%s"].memory.compareTo(quantity("2Gi")) >= 0`, driverA),
+							}},
+						),
+						subRequest(subReq1, classA, 1, resourceapi.DeviceSelector{
+							CEL: &resourceapi.CELDeviceSelector{
+								Expression: fmt.Sprintf(`device.capacity["%s"].memory.compareTo(quantity("1Gi")) >= 0`, driverA),
+							}},
+						),
+					),
+				),
+			),
+			classes: objects(class(classA, driverA)),
+			slices: objects(
+				slice(slice1, node1, pool1, driverA,
+					device(device1,
+						map[resourceapi.QualifiedName]resource.Quantity{
+							"memory": resource.MustParse("8Gi"),
+						},
+						map[resourceapi.QualifiedName]resourceapi.DeviceAttribute{
+							"driverVersion": {VersionValue: ptr.To("1.0.0")},
+						},
+					),
+				),
+				slice(slice2, node1, pool2, driverA,
+					device(device2,
+						map[resourceapi.QualifiedName]resource.Quantity{
+							"memory": resource.MustParse("2Gi"),
+						},
+						map[resourceapi.QualifiedName]resourceapi.DeviceAttribute{
+							"driverVersion": {VersionValue: ptr.To("2.0.0")},
+						},
+					),
+				),
+			),
+			node: node(node1, region1),
+
+			expectResults: []any{allocationResult(
+				localNodeSelector(node1),
+				deviceAllocationResult(req0, driverA, pool1, device1, false),
+				deviceAllocationResult(req1SubReq1, driverA, pool2, device2, false),
+			)},
+		},
+		"prioritized-list-subrequests-with-allocation-mode-all": {
```

Comment:
Oh, awesome, you have "all" already. Would be interesting though to have the claim structure I describe in the other comment, and make sure it behaves as we expect.

Reply:
Added.